vpc/tool/container2/setup_container_linux.go

```diff
@@ -745,25 +727,59 @@ func deleteFromIPv6BPFMap(ctx context.Context, assignment *vpcapi.AssignIPRespon
 	defer unix.Close(int(fd))
 	runtime.KeepAlive(path)
 
+	key := ipv6key(assignment)
 	uba := bpfAttrMapOpElem{
 		mapFd: uint32(fd),
-		key:   uint64(uintptr(unsafe.Pointer(&buf[0]))),
+		key:   uint64(uintptr(unsafe.Pointer(&key[0]))),
 	}
 	_, _, err = unix.Syscall(
 		unix.SYS_BPF,
 		unix.BPF_MAP_DELETE_ELEM,
 		uintptr(unsafe.Pointer(&uba)),
 		unsafe.Sizeof(uba),
 	)
-	if err != unix.ENOENT && err != 0 {
+	if err == unix.ENOENT {
+		err := errors.Wrap(err, "Unable to delete element for ipv6 map (notfound)")
+		logger.G(ctx).WithError(err).WithField("ip", assignment.Ipv6Address.Address.Address).WithField("classid", assignment.ClassId).WithField("vlanid", assignment.VlanId).Error("Element not found")
+		return err
+	} else if err != 0 {
 		logger.G(ctx).WithError(errors.Wrapf(err, "Unable to delete element for map with file descriptor %d", fd)).Error()
-		err2 := errors.Wrap(err, "Unable to delete element for ipv4 map")
+		err2 := errors.Wrap(err, "Unable to delete element for ipv6 map")
```

Comment:
Do we need the err2 pattern? Can you just return `errors.wrap` ?

Reply:
because err2 is an "error" whereas err is an syscall.Errno.